shared/actions/platform-specific/push.native.js

```diff
@@ -39,40 +39,20 @@ const updateAppBadge = (_, action) => {
 // If we're backgrounded we get the silent or the silent and real. To work around this we:
 // 1. Plumb through the intent from the java side if we relaunch due to push
 // 2. We store the last push and re-use it when this event is emitted to just 'rerun' the push
-let lastPushForAndroid = null
 const listenForNativeAndroidIntentNotifications = emitter => {
   const RNEmitter = new NativeEventEmitter(NativeModules.KeybaseEngine)
   // If android launched due to push
-  RNEmitter.addListener('androidIntentNotification', () => {
-    logger.info('[PushAndroidIntent]', lastPushForAndroid && lastPushForAndroid.type)
-    if (!lastPushForAndroid) {
+  RNEmitter.addListener('androidIntentNotification', (evt) => {
```

Comment:
Can you just do all this in Java?

Reply:
Not this part, this is the part that actually navigates once you open the app from the notification.

Re: the other part, also no, because the initial `newmessageSilent_2` is delivered not to our Java code but instead the `RNPushNotificationListenerService`, which then passes it to our JS. That could probably be refactored to completely cut out the `RNPushNotificationListenerService`, but it'll involve a lot more code.